packages/@lwc/ssr-compiler/src/compile-template/transformers/component/slotted-content.ts

```diff
@@ -36,51 +39,51 @@ import type {
-            let contentList = contentMap[name];
-            if (contentList) {
-                contentList.push(fn);
-            } else {
-                contentMap[name] = [fn];
-            }
-        }
+const bGenerateShadowSlottedContent = esTemplateWithYield`
+    const ${/* function name */ is.identifier} = (${/* local vars */ is.identifier}) => async function* ${/* function name */ 0}(contextfulParent) {
+        // The 'contextfulParent' variable is shadowed here so that a contextful relationship
+        // is established between components rendered in slotted content & the "parent"
+        // component that contains the <slot>.
+        ${/* shadow slot content */ is.statement}
+    };
+`<EsVariableDeclaration>;
+const bGenerateShadowSlottedContentRef = esTemplateWithYield`
+    const shadowSlottedContent = ${/* reference to hoisted fn */ is.identifier}(${/* local vars */ is.identifier});
+`<EsVariableDeclaration>;
+const bNullishGenerateShadowSlottedContent = esTemplateWithYield`
+    const shadowSlottedContent = null;
+`<EsVariableDeclaration>;
 
-        ${/* light DOM addLightContent statements */ is.expressionStatement}
-        ${/* scoped slot addLightContent statements */ is.expressionStatement}
+const bContentMap = esTemplateWithYield`
+    const ${/* name of the content map */ is.identifier} = Object.create(null);
+`<EsVariableDeclaration>;
+const bNullishContentMap = esTemplateWithYield`
+    const ${/* name of the content map */ is.identifier} = null;
+`<EsVariableDeclaration>;
+
+const bGenerateSlottedContent = esTemplateWithYield`
+    ${/* const shadowSlottedContent = ... */ is.variableDeclaration}
+    ${/* const lightSlottedContentMap */ is.variableDeclaration}
+    ${/* const scopedSlottedContentMap */ is.variableDeclaration}
+    ${/* light DOM addLightContent statements */ is.expressionStatement}
+    ${/* scoped slot addLightContent statements */ is.expressionStatement}
 `<EsStatement[]>;
 
 // Note that this function name (`__lwcGenerateSlottedContent`) does not need to be scoped even though
 // it may be repeated multiple times in the same scope, because it's a function _expression_ rather
 // than a function _declaration_, so it isn't available to be referenced anywhere.
 const bAddSlottedContent = esTemplate`
-    addSlottedContent(${/* slot name */ is.expression} ?? "", async function* __lwcGenerateSlottedContent(contextfulParent, ${
-        /* scoped slot data variable */ isNullableOf(is.identifier)
-    }, slotAttributeValue) {
-        ${/* slot content */ is.statement}
-    }, ${/* content map */ is.identifier});
+    addSlottedContent(
+        ${/* slot name */ is.expression} ?? "",
+        async function* __lwcGenerateSlottedContent(
+            contextfulParent,
+            ${/* scoped slot data variable */ isNullableOf(is.identifier)},
+            slotAttributeValue)
+        {
+            ${/* slot content */ is.statement}
+        },
+        ${/* content map */ is.identifier}
+    );
```

Comment:
With the placeholders, I found the previous version a bit difficult to read, whereas I found this a bit easier to read. However, if reviewers disagree, I'd be happy to revert this chunk.